### ISSUE TITLE
Got error in maml_train.py 

### ISSUE THREAD

**Issue by 688spiderbin** [Wenlin-Chen/FS-Mol-ADKF-IFT#9]
Hello, I get this error when I run though maml_train.py. 
When I try to test 
`valid_task_iterable = fsmol_dataset.get_task_reading_iterable(DataFold.VALIDATION)`
`task = next(iter(valid_task_iterable))`
 
It comes: 
![error_1](https://github.com/user-attachments/assets/281dd734-4c37-4eaa-b9dc-57f79899421c)
![error_2](https://github.com/user-attachments/assets/dde0c5cd-b7e9-465b-8d9b-24a63d73d1c2)

Could you tell me how to fix it ? 
I appreciate a lot !

**Comment by Wenlin-Chen** [comment]
Hi! I never encountered this error myself. Did you set up the environment following the instructions in the README file and put the data files to the correct directory? It could be that the package versions are different from what is required. Another possibility could be that there are Chinese characters in the PATH variable which the data loader cannot recognize. If that does not solve the problem, you could reach out to the authors of FS-Mol, since the MAML code was a fork from their repo as described in our README file.

**Comment by 688spiderbin** [comment]
Thanks for your suggestion. I fix it up by changing the DATA_PATH. 

And could you give me the specific version of "azureml-core", "botorch", "gpytorch" package. "azureml-core" related to "typing-extensions", which is easy to conflict with tensorflow. 
And "botorch", "gpytorch" related to the version of cuda. When I install them, my cuda 11 has been uninstall automatically and install cuda 12 due to the latest version of gpytorch. It's better to specify them otherwise the environment is messy and conflict.

Thank you !

**Comment by Wenlin-Chen** [comment]
Here are the package versions in my virtual environment:

- azureml-core 1.21.1
- pytorch 1.13.1
- botorch 0.6.2
- gpytorch 1.8.1
- typing-extensions 3.10.0.2
- tensorflow 2.8.0

If you still cannot resolve the conflicts, I would suggest using a different virtual environment for MAML. This is because the code for MAML was built upon tensorflow and doesn't rely on pytorch-related packages (at least not gpytorch or botorch) if I remember correctly. Hope this helps :)

**Comment by Wenlin-Chen** [comment]
Also, I think azureml-core was used for running experiments on the Azure server. If you don't use Azure, it might be that you don't need it at all.

**Comment by 688spiderbin** [comment]
Thanks for your patitent. Now I set up successfully for maml_train.py and protonet_train.py. 
Now is for par_train.py. I think I almost there and maybe one more question. What is the specific version of learn2learn package?

**Comment by Wenlin-Chen** [comment]
It's 0.1.7. It could also be 0.1.6 because I updated some packages later on and am not sure whether learn2learn got updated.

**Comment by 688spiderbin** [comment]
Thanks for your instructions, I think now I can run all the file successfully. But I noticed that each model running is quite time-consuming. How long did it take you to run each model on average ?

**Comment by Wenlin-Chen** [comment]
We were using a single NVIDIA GTX 2080Ti for the experiments. I remember our method ADKF-IFT took less than 12 hours to train and then a few hours to test. For some of the other methods like MAML, PAR, MAT, MT-GNN, etc, it could take a few days I think.

**Comment by 688spiderbin** [comment]
Hi, I got other problems, have you debug maml and protonet successfully ? I try to debug to figure out how they work.  I can run them but when it turn to debug mode it's very slow and seems to stuck. And I try to reduce the amount of training tasks, it comes to the error like this: 
 ![微信截图_20241014203020](https://github.com/user-attachments/assets/ac3caf13-8ca7-4851-98b1-1e2783dfde51)

I just wanna to output more validation metric through debugging, like auc,acc,pr,recall,f1 and cannot find the metric function. And also the code is maybe is old style and very complicated which have many generator, iter etc. Really hard to read and understand. 
Do you have any suggestion ?
Thank you very much. And my GPU is NVIDIA RTX 4090.

**Comment by 688spiderbin** [comment]
And when I change the parameter to get smaller size of of debugging,  my parameter changed to
--support_set_size=8
--query_set_size=16
--task_per_batch=2
--batch_size=8
--num_train_steps=1
--validate_every=1
--validation-support-set-size=[16,128]
--validation-query-set-size=12
--validation-num-samples=5
it comes up 
![微信截图_20241014234436](https://github.com/user-attachments/assets/5b18881b-57bd-480b-9a31-9e7d21d0141e)

While when i use your default parameter, this exception disappear. Maybe the parameter changing went something wrong ?

**Comment by AustinT** [comment]
Hello @688spiderbin , I think your questions are about the part of the code which we copied from FS-mol. We don't really know how to help. Perhaps post your issue there if you still have problems?

Since this original issue was about the versions of your python packages I will close the issue.
